src/fcmp_pp/curve_trees.cpp

```diff
@@ -1103,6 +1103,84 @@ template std::vector<crypto::ec_point> CurveTrees<Selene, Helios>::calc_hashes_f
     const typename CurveTrees<Selene, Helios>::Path &path,
     const bool replace_last_hash) const;
 //----------------------------------------------------------------------------------------------------------------------
+template<>
+CurveTreesV1::Path CurveTrees<Selene, Helios>::get_dummy_path(
+    const std::vector<fcmp_pp::curve_trees::OutputContext> &outputs,
+    uint8_t n_layers) const
+{
+    CHECK_AND_ASSERT_THROW_MES(m_c1_width == outputs.size(), "dummy path expects outputs == m_c1_width");
```

Comment:
Perhaps it could be good to be able to test when `outputs.size() < m_c1_width`. B/c `hash_grow` can work on a non-saturated layer of leafs, yeah? Or does it take "dummy" leaves when the layer is not complete?